site/docs/bin.md

```diff
@@ -53,26 +31,70 @@ The channel definition's `bin` property is for grouping quantitative, continuous
 }
 ```
 
-If `bin` is `true`, default binning parameters are used. To customize binning parameters, you can set `bin` to a bin definition object, which can have the following properties:
-
-
-{% include table.html props="maxbins,base,step,steps,minstep,divide,extent,nice" source="Bin" %}
+You can directly bin an `encoding` field by using the `bin` property in a [field definition](encoding.html#field).
 
+{% include table.html props="bin" source="PositionFieldDef" %}
 
-#### Examples
+{:#histogram}
+### Example: Histogram
 
-Given a field with quantitative continuous data values
+<!-- Given a field with quantitative continuous data values
 
 <div class="vl-example" data-name="point_1d"></div>
 
 Setting `bin` groups the values into a smaller number of bins.
 
-<div class="vl-example" data-name="point_1d_bin"></div>
+<div class="vl-example" data-name="point_1d_bin"></div> -->
 
 Mapping binned values and its count to a `bar` mark produces a histogram.
 
 <div class="vl-example" data-name="histogram"></div>
 
+{:#transform}
+## Bin Transform
+
+{: .suppress-error}
+```json
+// Any View Specification
+{
+  ...
+  "transform": [
+    {"bin": ..., "field": ..., "as" ...} // Bin Transform
+     ...
+  ],
+  ...
+}
+```
+
+The `bin` transform in the `transform` array has the following properties:
+
+{% include table.html props="bin,field,as" source="BinTransform" %}
+
+### Example: Histogram with Bin Transform
+
+Instead of using the `bin` property of a field definition, you can also a bin transform
+to derive a new field (e.g., `bin_IMDB_Rating`), and encode the new field instead.
+
+**TODO: example**
+
+<!--
```

Comment:
I'm okay with having this here as a reminder that this is broken. I will fix the issue soon.